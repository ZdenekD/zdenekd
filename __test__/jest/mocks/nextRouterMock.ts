const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        asPath: '/page/subpage/something-else',
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
