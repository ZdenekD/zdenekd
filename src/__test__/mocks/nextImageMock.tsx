const nextImageMock = jest.mock('next/image', () => function Image() {
    return <img src="" alt="" />;
});

export default nextImageMock;
