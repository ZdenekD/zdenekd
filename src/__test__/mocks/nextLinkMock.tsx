const nextLinkMock = jest.mock('next/link', () => function Link() {
    return <a href="/">link</a>;
});

export default nextLinkMock;
