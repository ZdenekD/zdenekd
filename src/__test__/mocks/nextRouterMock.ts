const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
