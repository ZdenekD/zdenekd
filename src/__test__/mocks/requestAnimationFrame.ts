import react from 'react';

window.requestAnimationFrame = () => 0;

export default react;
