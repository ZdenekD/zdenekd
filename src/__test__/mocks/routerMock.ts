const routerMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
    })),
}));

export default routerMock;
