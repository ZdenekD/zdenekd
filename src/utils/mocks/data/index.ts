import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
