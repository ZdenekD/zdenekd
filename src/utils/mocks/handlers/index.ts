import handler from './handler';

const handlers = [...handler];

export default handlers;
