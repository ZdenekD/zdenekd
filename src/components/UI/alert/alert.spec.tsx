import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '@/__test__/utils/helpers';
import '@/__test__/mocks/nextRouterMock';
import Alert from '.';

describe('UI/Alert', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Alert isOpen>message</Alert>);
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-alert');

        expect(component.exists()).toBe(true);
    });

    it('remove component on click remove button', () => {
        const mockSetState = jest.fn();

        React.useState = jest.fn(() => [true, mockSetState]);

        const container = shallow(<Alert>message</Alert>);
        const button = findComponent(container, 'component-alert-button');

        button.simulate('click');

        expect(mockSetState).toHaveBeenCalledTimes(1);
    });
});
