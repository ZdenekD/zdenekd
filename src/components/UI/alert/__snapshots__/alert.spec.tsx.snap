// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert match snapshot 1`] = `
<div class="backdrop "
     style="opacity:0"
>
</div>
<div class="alert  "
     data-test="component-alert"
     style="opacity:0;transform:scale(0.85) translateZ(0)"
>
  <section class="section">
    message
  </section>
  <button class="button"
          aria-label="Zavřít"
          data-test="component-alert-button"
  >
    ×
  </button>
</div>
`;
